### DISCUSSION TITLE
No data masks in rioxarray

### DISCUSSION THREAD

**Discussion by Scolymus** [corteva/rioxarray#610] [Q&A] [answered]
I'm creating a rio raster combining two channels (e.g., NDVI index map from sentinel-2 band images):

```
CH1 = riox.open_rasterio("path to file1").sel(band=1).astype('float32')
CH2 = riox.open_rasterio("path to file2").sel(band=1).astype('float32')
combo = ((CH1-CH2)/(CH1+CH2)).where((CH1+CH2)!=0., np.nan)
combo.rio.write_nodata(np.nan, inplace=True)
combo.rio.to_raster("/Users/ubuntu/image.tif")
```

However, I also want to save it in 8bit. Thus I do:

```
combo = (255*(combo + 1.0)/2.0).astype(np.uint8)
combo.rio.to_raster("/Users/ubuntu/image_8bit.tif")
```

Now my question is: When you convert the image to 8bit, some pixels are converted to 0, which is the same as NaN. Therefore you see gaps in the middle of the image. I saw [this solved for rasterio](https://rasterio.readthedocs.io/en/latest/topics/masks.html). However, I'm not sure how to do it in rioxarray and I wanted to avoid the parameter they fitted. I thought I could take the mask no_data from the combo before the 8bit conversion, fill the mask no_data in 8bit to everything is avalaible and then set the mask I took before. How can I do this?

 btw, why if I put combo.rio. in vscode I cannot see any property or method but if I write combo. I can see those for xarray?

**Comment by snowman2** [comment, marked as answer]
There currently isn't support for masks in rioxarray. This is due to how xarray handles integer masks:

https://github.com/pydata/xarray/issues/1194

You should be able to add the mask using rasterio after writing to disk with `r+` as mentioned in the documentation you linked.
